File: 2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.cpp
class Solution {
public:
    vector<vector<int>> findMatrix(vector<int>& nums) {
       int n=nums.size();
        
        vector<int> count(n+10);
        vector<vector<int>> result;
        
    for(int i=0; i<n;i++)
    {
        if(result.size()<=count[nums[i]])
        {
            result.push_back({});
            
        }
        result[count[nums[i]]++].push_back(nums[i]);
    }
        return result;
    }
};

File: 2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.java
class Solution {
    public List<List<Integer>> findMatrix(int[] nums) {
        ArrayList<List<Integer>> result= new ArrayList<>();
        
        int n=nums.length;
        int count[]= new int[n+1];
        
        for(int i=0; i<n;i++)
        {
            if(result.size()<=count[nums[i]])
            {
                result.add(new ArrayList<>());
                
            }
          //  count[nums[i]]++;
            result.get(count[nums[i]]++).add(nums[i]);
        }
        
        return result;
        
    }
}